spec/support/contexts/db_restore.rb

```diff
@@ -86,11 +86,40 @@ def set_spec_env_for_postgres_admin_basebackup_restore
       #   https://github.com/travis-ci/travis-build/blob/4f580b2/lib/travis/build/bash/travis_setup_postgresql.bash#L26-L27
       #   https://github.com/travis-ci/travis-cookbooks/blob/46a8e7fd/cookbooks/travis_postgresql/templates/ubuntu/9.5/postgresql.conf.erb#L41-L45
       #
-      ENV["APPLIANCE_PG_DATA"]    = "/var/lib/postgresql/9.5/main"
-      ENV["APPLIANCE_PG_SERVICE"] = "ci_pg_instance"
+      # Unfortunately the plot thickens a bit since they do two things that is
+      # different from our appliance installs:
+      #
+      #   - 'RAMFS' is used for their data directory, but is sync'd from
+      #     /var/lib/postgresql/9.5/main via their init scripts
+      #   - The configs and data are in separate dirs (/etc and /var/lib
+      #     respectively)
+      #
+      # An example of the running `postgres` command can be found below
+      #
+      #   $ /usr/lib/postgresql/9.5/bin/postgres                       \
+      #       -D /var/ramfs/postgresql/9.5/main                        \
+      #       -c config_file=/etc/postgresql/9.5/main/postgresql.conf
+      #
+      # The other issue we run into is that we basically require `sudo` for
+      # most of our actions in `PostgresAdmin`, and specs are run using the
+      # un-elevated CI user, travis.
+      #
+      # The command below basically runs the the PostgresAdmin command in a
+      # subprocess with elevated privleges, instead of trying to stub
+      # everything and correct it in a case by case basis.
+      env = {
+        "APPLIANCE_PG_DATA"    => "/var/lib/postgresql/9.5/main",
+        "APPLIANCE_PG_SERVICE" => "ci_pg_instance"
+      }
 
-      allow(LinuxAdmin::Service).to receive(:new).with(PostgresAdmin.service_name)
-                                                 .and_return(CiPostgresRunner)
+      dir_opts     = "-I #{RestoreHelper::SPEC_DIR} -I #{RestoreHelper::LIB_DIR}"
+      require_opts = "-r linux_admin -r ci_helper -r gems/pending/util/postgres_admin"
+      ruby_eval    = "-e 'PostgresAdmin.restore(#{RestoreHelper.restore_backup_opts.inspect})'"
+      cmd          = "sudo #{Gem.ruby} #{dir_opts} #{require_opts} #{ruby_eval}"
+
+      allow(PostgresAdmin).to receive(:restore).and_wrap_original do |_, *_args|
```

Comment:
Why use `and_wrap_original` here rather than `and_return` if you aren't using the yielded params?

Reply:
Few things with this (and I should probably add a comment around this:

First off, it is assumed with `PostgresAdmin` the we have `elevated` permissions to do most of the tasks in a restore.  I am even pretty sure that the `appliance_console` alias on an appliance is just a wrapper that includes calling `sudo /path/to/manageiq-appliance_console/bin/...` as well.  If I don't do this with `sudo`, basically all of the `FileUtils` commands will fail.

The commands do actually still run `PostgresAdmin.restore` with the same (see the `ruby_eval =` line above), but I just decided to pull the opts from the `RestoreHelper` instead of from the `*_args`.  Could have gone either way.

The other reason we use `and_wrap_original` is I wanted to do a `yield`, and not a static return that happened on execution of this line.  If I used return, it would have run this prior to it being called later in the spec.  This is probably pedantic, but it was mostly to do the best to simulate actually calling the method by instead call it through a elevated sub process with the limited number of files necessary to load `PostgresAdmin` and run `.restore`.

Reply:
Makes sense, but I think we should use the yielded args because otherwise what we pass into `.restore` in the test isn't actually being validated, right? I know it's just the same method result now, but I want to make sure this still works if that ends up changing for some reason.

Reply:
>  but I think we should use the yielded args...

No prob, and I was on the fence about this as well.  I will go ahead and change this.

Reply:
Turns out that there is a better way to do this as well:

https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/configuring-responses/block-implementation

So I will probably switch to using that as well.